src/galileo/middleware/tracing.py

```diff
@@ -0,0 +1,204 @@
+and makes them available to the Galileo logger within request handlers.
+
+Works with any ASGI framework built on Starlette:
+- FastAPI
+- Starlette
+- Any other Starlette-based framework
+
+Example usage with FastAPI:
+    ```python
+    from fastapi import FastAPI
+    from galileo.middleware import TracingMiddleware, get_request_logger
+
+    app = FastAPI()
+    app.add_middleware(TracingMiddleware)
+
+    @app.post("/process")
+    async def process_request(data: dict):
+        # Logger automatically continues the distributed trace
+        logger = get_request_logger()
+        logger.add_workflow_span(input=str(data), name="process_workflow")
+        # ... process request ...
+        logger.conclude(output="done")
+        return {"status": "success"}
+    ```
+
+Example usage with Starlette:
+    ```python
+    from starlette.applications import Starlette
+    from starlette.routing import Route
+    from galileo.middleware import TracingMiddleware, get_request_logger
+
+    async def homepage(request):
+        logger = get_request_logger()
+        logger.add_workflow_span(input="homepage", name="homepage_handler")
+        logger.conclude(output="success")
+        return {"status": "ok"}
+
+    app = Starlette(
+        routes=[Route("/", homepage)],
+        middleware=[TracingMiddleware]
+    )
+    ```
+"""
+
+import logging
+from typing import Optional
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.types import ASGIApp
+
+from galileo.constants.tracing import PARENT_ID_HEADER, TRACE_ID_HEADER
+from galileo.decorator import _parent_id_context, _trace_id_context
+from galileo.logger import GalileoLogger
+
+_logger = logging.getLogger(__name__)
+
+
+class TracingMiddleware(BaseHTTPMiddleware):
+    """
+    Middleware that extracts distributed tracing headers from incoming requests.
+
+    This middleware looks for X-Galileo-Trace-ID and X-Galileo-Parent-ID headers
+    in incoming HTTP requests and stores them in context variables, making them
+    available to request handlers via the `get_request_logger()` function.
+
+    The middleware is compatible with FastAPI and any Starlette-based framework.
+
+    Attributes
+    ----------
+    project : Optional[str]
+        Default project name for loggers created within requests
+    log_stream : Optional[str]
+        Default log stream name for loggers created within requests
+    """
+
+    def __init__(self, app: ASGIApp, project: Optional[str] = None, log_stream: Optional[str] = None) -> None:
+        """
+        Initialize the tracing middleware.
+
+        Parameters
+        ----------
+        app : ASGIApp
+            The ASGI application
+        project : Optional[str]
+            Default project name (can be overridden by GALILEO_PROJECT env var)
```

Comment:
Are you sure `GALILEO_PROJECT` is being used as the default when no value is set or to override the passed in value?